packages/hardhat-core/src/internal/hardhat-network/provider/vm/ethereumjs.ts

```diff
@@ -172,13 +172,28 @@ export class EthereumJSAdapter implements VMAdapter {
         }
       );
 
-      return await this._vm.runTx({
-        block: blockContext,
-        tx,
-        skipNonce: true,
-        skipBalance: true,
-        skipBlockGasLimitValidation: true,
-      });
+      const vmDebugTracer = new VMDebugTracer(this._vm);
+      let result: RunTxResult | undefined;
+      const trace = await vmDebugTracer.trace(
+        async () => {
+          result = await this._vm.runTx({
+            block: blockContext,
+            tx,
+            skipNonce: true,
+            skipBalance: true,
+            skipBlockGasLimitValidation: true,
+          });
+        },
+        {
+          disableStorage: true,
+          disableMemory: true,
+          disableStack: true,
```

Comment:
If you want extra information in the ethereumjs trace, set any (or all) of these values to `false`.